packages/aws-lambda/src/triggers.js

```diff
@@ -102,25 +107,43 @@ function extractHttpFromApiGatewwayProxyEvent(event, span) {
   if (event.version === '2.0') {
     const requestCtx = event.requestContext || {};
     const requestCtxHttp = requestCtx.http || {};
-
+    requestCtxHttp.path = {};
+    requestCtxHttp.rawPath = {};
     span.data.http = {
       method: requestCtxHttp.method,
-      url: requestCtxHttp.path,
-      path_tpl: event.rawPath,
+      url: normalizePath(requestCtxHttp.path),
+      path_tpl: normalizePath(event.rawPath),
       params: readHttpQueryParams(event),
       header: captureHeaders(event)
     };
   } else {
     span.data.http = {
       method: event.httpMethod,
-      url: event.path,
-      path_tpl: event.resource,
+      url: normalizePath(event.path),
+      path_tpl: normalizePath(event.resource),
       params: readHttpQueryParams(event),
       header: captureHeaders(event)
     };
   }
 }
 
+//  Ensures the `path` value is always a string. The backend expects `path` to be a string,
+//  so we convert any non-string (for example, an empty object) to null.
+//  For reference, see AWS documentation on the Lambda event input format for API Gateway and Function URLs:
+//  eslint-disable-next-line max-len
+//  https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html#api-gateway-simple-proxy-for-lambda-input-format
+//  https://docs.aws.amazon.com/lambda/latest/dg/urls-invocation.html#urls-payloads
+function normalizePath(value) {
+  if (typeof value === 'string') {
+    return value;
+  }
+  logger.debug(
+    `Received a non-string value for the "path" field from the incoming request event: ${JSON.stringify(value)}. ` +
+      'Expected a string. This may result in incomplete or broken traces in the Instana UI.'
```

Comment:
Can we please add an official aws link? That its clear that its not our requirement.

Reply:
We need to make clear that hit is an AWS misconfiguration.